simplified-books-controller/src/main/java/org/nypl/simplified/books/controller/ProfileAccountLogoutTask.kt

```diff
@@ -140,6 +158,13 @@ class ProfileAccountLogoutTask(
     }
 
     this.debug("device is activated and DRM is supported, running deactivation")
+    val token = this.handlePatronUserProfile()
+    if (token == null) {
+      this.warn("Patron user profile contained no Adobe DRM client token")
+      val message = "Patron user profile is missing DRM information."
+      this.steps.currentStepFailed(message, "patronUserProfileNoDRM")
+      throw IOException(message)
```

Comment:
It's possible for a library to exist in the system but not offer any Adobe client token functionality. We don't have any libraries like that right now (apart from open-access libraries, which don't have "logout") but we will in the future. Is this going to prevent "logout" from working for that sort of library?

Reply:
Yes, it might. This code is due some heavy refactoring like the borrow operation recently received. This'll need to be folded into that refactoring work.